src/screens/factoryPipelines/factoryPipelines.tsx

```diff
@@ -0,0 +1,174 @@
+import { toast } from "sonner";
+import { Experimental_CrudDataTable } from "@/widgets";
+import { ColumnDef } from "@tanstack/table-core";
+import { Factory } from "@/entities/factory";
+import {
+  TwinClass_DETAILED,
+  TwinClassResourceLink,
+} from "@/entities/twinClass";
+import { Check } from "lucide-react";
+import { TwinClassStatusResourceLink, TwinStatus } from "@/entities/twinStatus";
+import { FactoryResourceLink } from "@/entities/factory/components/resource-link/resource-link";
+
+const colDefs: Record<
+  keyof Omit<
+    FactoryPipeline,
+    | "inputTwinClassId"
+    | "factoryId"
+    | "factoryConditionSetId"
+    | "nextFactoryId"
+    | "outputTwinStatusId"
+  >,
+  ColumnDef<FactoryPipeline>
+> = {
+  id: {
+    id: "id",
+    accessorKey: "id",
+    header: "ID",
+    cell: (data) => <GuidWithCopy value={data.getValue<string>()} />,
+  },
+  description: {
+    id: "description",
+    accessorKey: "description",
+    header: "Description",
+  },
+  factory: {
+    id: "factory",
+    accessorKey: "factory",
+    header: "Factory",
+    cell: ({ row: { original } }) => (
+      <div className="max-w-48 inline-flex">
+        {original.factory && (
+          <FactoryResourceLink data={original.factory as Factory} withTooltip />
+        )}
+      </div>
+    ),
+  },
+  // TODO: Replace with a condition set resource link
+  factoryConditionSet: {
+    id: "factoryConditionSet",
+    accessorKey: "factoryConditionSet",
+    header: "Factory Condition Set",
+    cell: ({ row: { original } }) => (
+      <span>{original.factoryConditionSet?.name}</span>
+    ),
+  },
+  factoryConditionSetInvert: {
+    id: "factoryConditionSetInvert",
+    accessorKey: "factoryConditionSetInvert",
+    header: "Factory Condition Set Invert",
+    cell: (data) => <>{data.getValue() && <Check />}</>,
+  },
+  inputTwinClass: {
+    id: "inputTwinClass",
+    accessorKey: "inputTwinClass",
+    header: "Input Twin Class",
+    cell: ({ row: { original } }) => (
+      <div className="max-w-48 inline-flex">
+        <TwinClassResourceLink
+          data={original.inputTwinClass as TwinClass_DETAILED}
+          withTooltip
+        />
+      </div>
+    ),
+  },
+  outputTwinStatus: {
+    id: "outputTwinStatus",
+    accessorKey: "outputTwinStatus",
+    header: "Output Twin Status",
+    cell: ({ row: { original } }) =>
+      original.outputTwinStatus && (
+        <div className="max-w-48 inline-flex">
+          <TwinClassStatusResourceLink
+            data={original.outputTwinStatus as TwinStatus}
+            twinClassId={original.inputTwinClassId!}
+            withTooltip
+          />
+        </div>
+      ),
+  },
+  nextFactory: {
+    id: "nextFactory",
+    accessorKey: "nextFactory",
+    header: "Next Factory",
+    cell: ({ row: { original } }) => (
+      <div className="max-w-48 inline-flex">
+        {original.nextFactory && (
+          <FactoryResourceLink
+            data={original.nextFactory as Factory}
```

Comment:
need to add `hydration` to `searchFactoryPipelines` so we can remove `as Factory`